packages/astro/src/runtime/server/render/util.ts

```diff
@@ -224,3 +229,12 @@ export function promiseWithResolvers<T = any>(): PromiseWithResolvers<T> {
 		reject,
 	};
 }
+
+function urlCanParse(url: string) {
+	try {
+		new URL(url);
```

Comment:
Are relative URLs expected to be unescaped as well? If the OG image in the example were only `/api/og?title=hello&description=somedescription` to be relative to the current browser origin, it wouldn't be escaped as that is not a valid URL.

If we want to validate relative URLs as well this should be:
```suggestion
		new URL(url, 'http://example.com');
```

Any base there would work.

Reply:
URL.canParse wouldn't have allowed it but this seems like a great enhancement to me, thanks!

Reply:
On second thought, it may be too permissive.
I don't think there's a case anymore where urlCanParse will fail. If the attribute value has a "&", it always prints unescaped.

Reply:
`URL.canParse` can also receive a base as a second argument, so this applies after upgrading the Node version as well

Reply:
You're right, and we might want to be careful with it as well. I'm not sure how valid it is, but the threat is that we let an attribute like "&\\"" through and break everything.

Reply:
> On second thought, it may be too permissive. I don't think there's a case anymore where urlCanParse will fail. If the attribute value has a "&", it always prints unescaped.

Indeed, it is very hard for a string not to be a valid URL when there is a base. It was hard to find a test case that didn't pass it when we had a similar use case.

Makes this check do almost nothing.

Reply:
You're right. Undoing the change.